hr_timesheet_employee_allowed/models/account_analytic_line.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    @api.model
+    def default_get(self, field_list):
+        res = super(AccountAnalyticLine, self).default_get(field_list)
+        if res.get('employee_id'):
+            res.pop('employee_id', None)
+        return res
+
+    timesheet_line_user_id = fields.Many2one(
```

Comment:
Don't we already have a user field on TS lines?

Reply:
Yes we have `user_id` field in base. But we want to keep the current logged-in user in the line to track the user. It will be change based on the employee if I use `user_id` field. It is more useful to handle the domain.